src/substrait/textplan/converter/ReferenceNormalizer.cpp

```diff
@@ -14,27 +14,19 @@ namespace {
 bool compareExtensionFunctions(
     const ::substrait::proto::extensions::SimpleExtensionDeclaration& a,
     const ::substrait::proto::extensions::SimpleExtensionDeclaration& b) {
-  // First sort so that extension functions proceed any other kind of extension.
-  if (a.has_extension_function() && !b.has_extension_function()) {
-    return true;
-  } else if (!a.has_extension_function() && b.has_extension_function()) {
-    // Extension functions always come first.
-    return false;
-  } else if (!a.has_extension_function() && !b.has_extension_function()) {
-    // Both are not extension functions, no difference in ordering.
-    return false;
-  }
-  // Now sort by space.
-  if (a.extension_function().extension_uri_reference() <
-      b.extension_function().extension_uri_reference()) {
-    return true;
-  } else if (
-      a.extension_function().extension_uri_reference() >
-      b.extension_function().extension_uri_reference()) {
-    return false;
-  }
-  // Finally sort by name within a space.
-  return a.extension_function().name() < b.extension_function().name();
+  auto ord = [](const auto& decl) {
+    return make_tuple(
+        // First sort so that extension functions precede any other kind of
+        // extension.
+        not decl.has_extension_function(),
+        // Next sort by space.
+        decl.extension_function().extension_uri_reference(),
+        // Finally sort by name within a space.
+        decl.extension_function().name());
```

Comment:
Using `make_tuple` here will create copies of the strings on each comparison. We can avoid this by wrapping the strings in `string_view`
```suggestion
    return make_tuple(
        // First sort so that extension functions precede any other kind of
        // extension.
        not decl.has_extension_function(),
        // Next sort by space.
        std::string_view{decl.extension_function().extension_uri_reference()},
        // Finally sort by name within a space.
        std::string_view{decl.extension_function().name()});
```